src/utils/src/Math.c

```diff
@@ -0,0 +1,30 @@
+#include "Include_i.h"
+
+/**
+ * Calculates power using Divide and Conquer style - O(log exponent) complexity
+ *
+ * WARNING:
+ * The result could overflow if the base ^ power exceeds UINT64 limit.
+ * It is the callers responsibility to sanitize the values before calling this API
+ */
+UINT64 power(UINT32 base, UINT32 exponent) {
+    UINT64 result = 1;
+
+    if (base == 0) {
+        return 0;
+    }
+
+    // loop till `exponent becomes 0
```

Comment:
I strongly suggest getting rid of this comment which doesn't seem to provide any value and instead add a comment describing the algorithm. I am not entirely sure why the choice of this particular algo as it's not immediately clear (simple perf might be the reason but not obvious)

Reply:
It's not obvious I agree. But it is faster relative to the previous implementation. I've added a comment on the top indicating it's DAC based algorithm and the complexity.